python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_structured_config_types.py

```diff
@@ -300,3 +302,136 @@ def a_job():
         {"ops": {"a_struct_config_op": {"config": {"my_dict": {keys[0]: 1, keys[1]: 2}}}}}
     )
     assert executed["yes"]
+
+
+def test_descriminated_unions():
+    class Cat(Config):
+        pet_type: Literal["cat"]
+        meows: int
+
+    class Dog(Config):
+        pet_type: Literal["dog"]
+        barks: float
+
+    class Lizard(Config):
+        pet_type: Literal["reptile", "lizard"]
```

Comment:
should we even allow multiple literals like this? I think that forcing users to add a Reptile class here might be more clear.

Reply:
Not sure we should advertise support, but this is part of Pydantic's spec so I think it can't hurt to support it. It doesn't really affect our implementation.